core/iwasm/aot/aot_runtime.c

```diff
@@ -1566,13 +1568,15 @@ aot_module_free(AOTModuleInstance *module_inst, uint32 ptr)
             char *free_func_name;
 
             if (module->retain_func_index != (uint32)-1) {
-                free_func_name = "__unpin";
+                free_func_name = "__release";
             }
             else {
                 free_func_name = "free";
             }
             free_func =
                 aot_lookup_function(module_inst, free_func_name, "(i)i");
+            if (!free_func)
```

Comment:
Should add a condition:
```C
if (!free_func && module->retain_func_index != (uint32)-1)
```

Reply:
Hi @wenyongh, It would be more reasonable to add the condition and I will modify it. Could you please tell me when free_func_name is "free"? Does it exist an older version of AssemblyScript before 0.17?

Reply:
Hi, the "free" function can be exported by wasi-sdk or emsdk for wasm application written by C language, it isn't exported by asc yet. Here retain_func_index is only set for asc compiler when both "__new" and "__retain" (or "__pin") functions are exported.